x-pack/platform/plugins/shared/fleet/server/routes/cloud_connector/handlers.ts

```diff
@@ -25,6 +27,23 @@ import type {
   DeleteCloudConnectorRequestSchema,
 } from '../../types/rest_spec/cloud_connector';
 
+const validateCreateCloudConnectorRequest = (
+  body: TypeOf<typeof CreateCloudConnectorRequestSchema.body>
+): CreateCloudConnectorRequest => {
+  // Runtime check for AWS
+  if (body.cloudProvider === 'aws') {
```

Comment:
Duplicate check we are already checking for `validateCloudConnectorDetails`. Let's keep the PR focus refactoring the front-end Phase 1  Azure Cloud Connector code into `CloudConnnectorComponent`. We are making changes to code where we already have check for validation.